forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -0,0 +1,326 @@
+
+                let month_string = format!("{:02}", now.month() as u8);
+                let year = now.year();
+                let day_string = format!("{:02}", now.day() as u8);
+                let chain_name = chain_get_name().await.map_err(handle_rpc_err).unwrap();
+
+                let vars = HashMap::from([
+                    ("year".to_string(), year.to_string()),
+                    ("month".to_string(), month_string),
+                    ("day".to_string(), day_string),
+                    ("chain".to_string(), chain_name),
+                    ("height".to_string(), epoch.to_string()),
+                ]);
+                let output_path = match strfmt(&output_path.display().to_string(), &vars) {
+                    Ok(path) => path.into(),
+                    Err(e) => {
+                        cli_error_and_die(format!("Unparsable string error: {}", e), 1);
+                    }
+                };
+
+                let params = (
+                    epoch,
+                    *recent_stateroots,
+                    *include_old_messages,
+                    output_path,
+                    TipsetKeysJson(chain_head.key().clone()),
+                    *skip_checksum,
+                );
+
+                // infallible unwrap
+                let out = chain_export(params).await.map_err(handle_rpc_err).unwrap();
+
+                println!("Export completed. Snapshot located at {}", out.display());
+            }
+            Self::Fetch { snapshot_dir } => {
+                let snapshot_dir = snapshot_dir
+                    .clone()
+                    .unwrap_or_else(|| default_snapshot_dir(&config));
+                match snapshot_fetch(&snapshot_dir, config).await {
+                    Ok(out) => println!("Snapshot successfully downloaded at {}", out.display()),
+                    Err(e) => cli_error_and_die(format!("Failed fetching the snapshot: {e}"), 1),
+                }
+            }
+            Self::Dir => {
+                let dir = default_snapshot_dir(&config);
+                println!("Default snapshot dir: {:?}", dir);
+            }
+            Self::List { snapshot_dir } => {
+                let snapshot_dir = snapshot_dir
+                    .clone()
+                    .unwrap_or_else(|| default_snapshot_dir(&config));
+                println!("Snapshot dir: {:?}", snapshot_dir);
+                if let Ok(dir) = fs::read_dir(snapshot_dir) {
+                    println!("\nLocal snapshots:");
+                    for entry in dir.flatten() {
+                        let path = entry.path();
+                        if path.is_file() {
+                            if let Some(filename) = entry.file_name().to_str() {
+                                if filename.ends_with(".car") {
+                                    println!("{filename}");
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
This snippet is a bit complex, it would be nice to move it to a separate, private method (in general, for readability it would be even better to have all those commands in different methods).

Another thing, it may become more readable if you'd a nice combo along the lines of:
```
fs::read_dir(snapshot_dir)
   .flatten()
   .map(|entry| entry.path())
   .filter(|entry| entry.is_file())
   ...
```
or maybe it wouldn't be more readable. Up to you :)

Reply:
fixed